lib/archethic/p2p/geo_patch.ex

```diff
@@ -24,73 +24,44 @@ defmodule ArchEthic.P2P.GeoPatch do
   end
 
   defp compute_random_patch do
-    list_char = Enum.concat([?0..?9, ?A..?F])
-    Enum.take_random(list_char, 3) |> List.to_string()
+    list_char1 = Enum.concat([?0..?9, ?A..?F])
+    list_char2 = Enum.concat([?0..?3, ?C..?F])
+
+    Enum.take_random(list_char1, 2)
+    |> List.insert_at(1, Enum.take_random(list_char2, 1))
+    |> List.to_string()
   end
 
   defp compute_patch(lat, lon) do
-    lat_sign = sign(lat)
-    lon_sign = sign(lon)
-
-    fdc = [lat / 90, lon / 180]
-
-    sd =
-      [(lat - lat_sign * 45) / 2, (lon - lon_sign * 90) / 2]
-      |> resolve_with_sign([lat, lon])
+    lon_pos = (lon + 180) / 22.5
+    # Adding 4 to have second digit hex value from C to 3
+    lat_pos = (lat + 90) / 22.5 + 4
```

Comment:
avoid adding 4 and adding 90, 180 was in earlier implementation so, avoid it

Reply:
Hello ! 
I don't really understand your suggestion. Here is why I choose this implementation instead of the one did before : 
The first calculations with adding 90 / 180 / 4 result in an index between 0 and 15 for ```lon_pos``` one and between 4 and 11 for ```lat_pos```.
With these index, we can easily find the good hexadecimal value in the array of the function ```main_index_patch```.

The advantage of this implementation is that it takes less ressources and less execution times than the previous implementation, i.e guetting a value from an array by his index is faster than doing multiple if condition to find the good hexadecimal value.
An other advantage is that it takes less lines of code.

To be easier to read, instead of adding 4 to ```lat_pos``` we can create a new array [C,D,E,F,0,1,2,3] and get hexadecimal value from it, but I think that using same array as for ```lon_pos``` is better to reduce memory usage.

@samuel-uniris What is your advice about it ?

Reply:
So in term of performance pattern matching (multiple function clause) is like if/else block under the good bu is quite fast and optimized during compilation.

Accessing a random element in a list is not the same in Erlang. List are not arrays but linked list , which means to access nth element, need to scan nth before . When the list is big is not efficient. 
However if the list is bounded, know and static, you can leverage tuple where access a specific element is fast. 

There is different strategy for this problem. 

In term of readability the pattern matching is better when you have a several if/else/case blocks.

Reply:
So in term of performance pattern matching (multiple function clause) is like if/else block under the good bu is quite fast and optimized during compilation.

Accessing a random element in a list is not the same in Erlang. List are not arrays but linked list , which means to access nth element, need to scan nth before . When the list is big is not efficient. 
However if the list is bounded, know and static, you can leverage tuple where access a specific element is fast. 

There is different strategy for this problem. 

In term of readability the pattern matching is better when you have a several if/else/case blocks.

Reply:
Hi ! Updated in last commit to use tuple. Using pattern matching function was not as easy to read because it needs to much function to catch all conditions.